airbyte-ci/connectors/live-tests/src/live_tests/commons/backends/file_backend.py

```diff
@@ -22,101 +30,57 @@ class FileBackend(BaseBackend):
-            with open(f"{self._output_directory}/{stream}_{self.RELATIVE_STATES_PATH}", "w") as states_file:
-                for state in states:
-                    states_file.write(f"{state.json()}\n")
-
-    @staticmethod
-    def _get_messages_by_type(messages: Iterable[AirbyteMessage]) -> Dict[str, Union[List, Dict]]:
-        messages_by_type: Dict[str, Union[List, Dict]] = {
-            "catalog": [],
-            "connection_status": [],
-            "records": defaultdict(list),
-            "spec": [],
-            "states": defaultdict(list),
-            "traces": [],
-            "logs": [],
-            "controls": [],
-        }
-
-        for message in messages:
-            if not isinstance(message, AirbyteMessage):
-                continue
-
-            if message.type == AirbyteMessageType.CATALOG:
-                assert isinstance(messages_by_type["catalog"], list)
-                messages_by_type["catalog"].append(message.catalog)
-
-            elif message.type == AirbyteMessageType.CONNECTION_STATUS:
-                assert isinstance(messages_by_type["connection_status"], list)
-                messages_by_type["connection_status"].append(message.connectionStatus)
-
-            elif message.type == AirbyteMessageType.RECORD:
-                assert isinstance(messages_by_type["records"], dict)
-                messages_by_type["records"][message.record.stream].append(message.record)
-
-            elif message.type == AirbyteMessageType.SPEC:
-                assert isinstance(messages_by_type["spec"], list)
-                messages_by_type["spec"].append(message.spec)
-
-            elif message.type == AirbyteMessageType.STATE:
-                if message.state.stream and message.state.stream.stream_descriptor:
-                    stream_name = message.state.stream.stream_descriptor.name
-                    stream_namespace = message.state.stream.stream_descriptor.namespace
-                    key = f"{stream_name}_{stream_namespace}" if stream_namespace else stream_name
-                else:
-                    key = "_global_states"
-                assert isinstance(messages_by_type["states"], dict)
-                messages_by_type["states"][key].append(message.state)
-            elif message.type == AirbyteMessageType.TRACE:
-                assert isinstance(messages_by_type["traces"], list)
-                messages_by_type["traces"].append(message.trace)
-            elif message.type == AirbyteMessageType.LOG:
-                assert isinstance(messages_by_type["logs"], list)
-                messages_by_type["logs"].append(message.log)
-            elif message.type == AirbyteMessageType.CONTROL:
-                assert isinstance(messages_by_type["controls"], list)
-                messages_by_type["controls"].append(message.control)
-        return messages_by_type
+    async def write(self, airbyte_messages: Iterable[AirbyteMessage]):
```

Comment:
Does this really have to be async? If I'm not mistaken the caller is not using any concurrent call / task group so it'll remain a blocking operation.